File: 345reverseVowelsOfAString/reverseVowelsOfAString.java
class Solution {
    public String reverseVowels(String s) {
        final var matcher = java.util.regex.Pattern.compile("[aeiou]", java.util.regex.Pattern.CASE_INSENSITIVE).matcher(s);
        final var vowels = matcher.results().map(group -> group.group()).collect(Collectors.toCollection(LinkedList::new));
        Collections.reverse(vowels);
        matcher.reset();
        return matcher.replaceAll(group -> vowels.pop());
    }
}

File: 345reverseVowelsOfAString/reverseVowelsOfAString.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function(s) {
    const vowels = Array.from(s.matchAll(/[aeiou]/gi), group => group[0])
    return s.replace(/[aeiou]/gi, group => vowels.pop())
};
